crates/pubsub/src/managers/in_flight.rs

```diff
@@ -13,26 +13,31 @@ pub(crate) struct InFlight {
     pub(crate) request: SerializedRequest,
 
     /// The channel to send the response on.
-    pub(crate) tx: oneshot::Sender<Result<Response, TransportError>>,
+    pub(crate) tx: Option<oneshot::Sender<Result<Response, TransportError>>>,
```

Comment:
instead of changing the type here, you can just make a broken sender with `let (tx, _) = oneshot::channel();`

Reply:
or use `Inflight::new` and then drop the receiver/waiter